src/services/photos/network/PhotosNetworkManager.ts

```diff
@@ -144,6 +147,11 @@ export class PhotosNetworkManager implements RunnableService<PhotosNetworkManage
    */
   public async processUploadOperation(operation: PhotosNetworkOperation): Promise<OperationResult | null> {
     this.onUploadStartCallback(operation.photosItem);
+    const stopIfAborted = () => {
+      if (this.isAborted) {
```

Comment:
if is already aborted, why throw the abort error again? Is [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) available in RN? This pattern is browser-native and provides a signal handled internally to check if is already been aborted.

Reply:
yes is available, should move this to `abortController.signal.aborted`

Could happen that when we abort the system, and empty the queue, a task will be still being processed, for that case this is used, since not all the operations support aborts. For example the generate preview operation cannot be aborted, but we can check if the system is aborted before commiting